### ISSUE TITLE
HPC-GAP: SerializeToNativeString broken on records

### ISSUE THREAD

**Issue by ChrisJefferson** [gap-system/gap#3259]
```
gap> r := rec(a := 1, b := 2, c := 3, d := 4, e := 5);
gap> SerializeToNativeString(r);
```

Segfaults. I haven't traced the whole history, but it's not clear to me how this ever could have worked. The initial problem is it's missing that `GET_RNAM_PREC` can return a negative value, but if anyone was going to fix this, I'd hope they would test it fairly carefully.

**Comment by fingolfin** [comment]
I am not surprised that it doesn't work. I spent some time on that serialization and traversal code, and fixed several bugs that really looked as if that code had never been executed. On the up side, in each case it was fairly easy to fix it.

**Comment by fingolfin** [comment]
This works fine now, I believed I've fixed it in 10a05b2f2c931c8fbb52192e0d5cccf873f98740 but forgot to close this issue.
